src/db.h

```diff
@@ -145,6 +145,7 @@ struct db_filter_snap {
 	struct db_filter **filters;
 	unsigned int filter_cnt;
 	bool shadow;
+	bool user;
```

Comment:
I'm thinking a `bool` should meet our needs initially, but I'm wondering if long-term we'll want to switch to a transaction number (with a special bit for user-initiated transactions).  I'm afraid the `bool` check of `if snap->user != user` may not be powerful enough someday.

The good thing is none of this exposed by the API, so we could change it later.

Reply:
Agree, let's leave it as a bool now, mostly for simplicity; we can change it later if needed.